platforms/iOS/vm/OSX/sqSqueakOSXMetalView.m

```diff
@@ -0,0 +1,782 @@
+	if(metalInitialized)
+		return;
+	metalInitialized = YES;
+	if(self.device == nil)
+		self.device = MTLCreateSystemDefaultDevice();
+		
+	[self buildPipelines];
+	[self createScreenQuad];
+
+	// Create the command queue.
+	graphicsCommandQueue = [self.device newCommandQueue];
+}
+
+- (void) buildPipelines {
+	NSError *libraryError;
+	dispatch_data_t metalLibraryData = dispatch_data_create(SqueakMainShaders_metallib, SqueakMainShaders_metallib_len, dispatch_get_global_queue(0, 0), ^{});
+	id<MTLLibrary> shaderLibrary = [self.device newLibraryWithData: metalLibraryData error: &libraryError];
+#if !__has_feature(objc_arc)
+	dispatch_release(metalLibraryData);
+#endif
+	if(!shaderLibrary)
+	{
+		NSLog(@"Shader library error: %@", libraryError.localizedDescription);
+		return;
+	}
+	
+	// Retrieve the shaders from the shader libary.
+	id<MTLFunction> vertexShader = [shaderLibrary newFunctionWithName: @"screenQuadFlipVertexShader"];
+	id<MTLFunction> fragmentShader = [shaderLibrary newFunctionWithName: @"screenQuadFragmentShader"];
+	if(!vertexShader || !fragmentShader)
+	{
+		RELEASEOBJ(shaderLibrary);
+		return;
+	}
+
+	// Create the screen quad pipeline.
+	MTLRenderPipelineDescriptor *pipelineDescriptor = [MTLRenderPipelineDescriptor new];
+	pipelineDescriptor.vertexFunction = vertexShader;
+	pipelineDescriptor.fragmentFunction = fragmentShader;
+	pipelineDescriptor.inputPrimitiveTopology = MTLPrimitiveTopologyClassTriangle;
+	pipelineDescriptor.colorAttachments[0].pixelFormat = self.colorPixelFormat;
+	
+	NSError *pipelineError = NULL;
+	screenQuadPipelineState = [self.device newRenderPipelineStateWithDescriptor: pipelineDescriptor error: &pipelineError];
+	RELEASEOBJ(shaderLibrary);
+	RELEASEOBJ(vertexShader);
+	RELEASEOBJ(fragmentShader);
+	if(!screenQuadPipelineState)
+	{
+		NSLog(@"Pipeline state creation error: %@", pipelineError.localizedDescription);
+		return;
+	}
+}
+
+- (void) createScreenQuad {
+	screenQuadVertexBuffer = [self.device
+		newBufferWithBytes: screenQuadVertices
+		length: sizeof(screenQuadVertices)
+		options: MTLResourceStorageModeManaged];
+}
+
+-(void)drawRect:(NSRect)rect
+{
+	[self setupMetal];
+    [self setupFullScreendispBitsIndex];
+
+	// Alawys try to fill the texture with the pixels.
+	if ( fullScreendispBitsIndex ) {
+		[self loadTexturesFrom: fullScreendispBitsIndex subRectangle: (clippyIsEmpty ? rect : NSRectFromCGRect(clippy))];
+		//[self loadTexturesFrom: fullScreendispBitsIndex subRectangle: rect];
+		clippyIsEmpty = YES;
+	    syncNeeded = NO;
+	}
+	
+	MTLRenderPassDescriptor *renderPassDescriptor = self.currentRenderPassDescriptor;
+	if(renderPassDescriptor != nil && self.currentDrawable)
+	{
+		currentCommandBuffer = [graphicsCommandQueue commandBuffer];
+		currentRenderEncoder = [currentCommandBuffer renderCommandEncoderWithDescriptor: renderPassDescriptor];
+		
+		// Set the viewport.
+		[currentRenderEncoder setViewport: (MTLViewport){0.0, 0.0, lastFrameSize.size.width, lastFrameSize.size.height}];
```

Comment:
When I change this line to
```c
[currentRenderEncoder setViewport: (MTLViewport){0.0, 0.0, lastFrameSize.size.width * 2, lastFrameSize.size.height * 2}];
```

the problem I mentioned in https://github.com/OpenSmalltalk/opensmalltalk-vm/pull/306#issuecomment-453290159 goes away. But this is probably not the right fix.

Reply:
Hi, sorry for taking too long. I was completely busy with my job. I just changed the code to use the size from  the drawableSurface property, instead of the size of the window. Can you check whether this fixes the issue or not?